Piped crashes with a nil pointer problem

**What happened**:

I have deployed some applications with PipeCD, but this started to crash at some point.
Piped crashes with a nil pointer problem. The stack trace is as follows.

```
{"severity":"INFO","eventTime":1604254142.5224824,"logger":"piped.piped.livestatestore.kubernetes.reflector","caller":"kubernetes/reflector.go:228","message":"all informer caches have been synced","serviceContext":{"service":"piped.piped","version":"v0.8.0"},"cloud-provider":"kubernetes-default"}
{"severity":"INFO","eventTime":1604254142.5224955,"logger":"piped.piped.livestatestore.kubernetes","caller":"kubernetes/kubernetes.go:116","message":"the reflector has done the first sync","serviceContext":{"service":"piped.piped","version":"v0.8.0"},"cloud-provider":"kubernetes-default"}
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x16090cc]

goroutine 51 [running]:
github.com/pipe-cd/pipe/pkg/app/piped/livestatestore/kubernetes.(*appNodes).addDependedResource(0x0, 0xc000b8b020, 0x24, 0xc000052534, 0x2, 0xc000052528, 0x3, 0xc0004ad849, 0x7, 0xc000be48a0, ...)
        pkg/app/piped/livestatestore/kubernetes/appnodes.go:113 +0x17c
github.com/pipe-cd/pipe/pkg/app/piped/livestatestore/kubernetes.(*store).initialize(0xc000079560)
        pkg/app/piped/livestatestore/kubernetes/store.go:74 +0x321
github.com/pipe-cd/pipe/pkg/app/piped/livestatestore/kubernetes.(*Store).Run(0xc00035a280, 0x1d110c0, 0xc0003b2480, 0x12, 0xc0002083e8)
        pkg/app/piped/livestatestore/kubernetes/kubernetes.go:118 +0x5a3
github.com/pipe-cd/pipe/pkg/app/piped/livestatestore.(*store).Run.func1(0x0, 0x0)
        pkg/app/piped/livestatestore/livestatestore.go:126 +0x87
golang.org/x/sync/errgroup.(*Group).Go.func1(0xc000924720, 0xc00092c3a0)
        external/org_golang_x_sync/errgroup/errgroup.go:57 +0x59
created by golang.org/x/sync/errgroup.(*Group).Go
        external/org_golang_x_sync/errgroup/errgroup.go:54 +0x66
```

**What you expected to happen**:

The piped works normally. In addition, if a nil pointer problem occurs, the piped should be properly recovered.

**How to reproduce it**:

I have no idea for now.

**Environment**:
- `piped` version: `v0.8.0`
- `control-plane` version: `v0.8.0`
- Others:

--- a/pkg/app/piped/livestatestore/kubernetes/store.go
+++ b/pkg/app/piped/livestatestore/kubernetes/store.go
@@ -151,7 +151,6 @@ func (s *store) addResource(obj *unstructured.Unstructured, appID string) {
 	s.mu.Lock()
 	s.resources[uid] = appResource{appID: appID, owners: owners, resource: obj}
 	s.mu.Unlock()
-
 }
 
 func (s *store) onAddResource(obj *unstructured.Unstructured) {
@@ -202,7 +201,6 @@ func (s *store) onDeleteResource(obj *unstructured.Unstructured) {
 	}
 
 	// Handle depended nodes from here.
-
 	if appID == "" {
 		s.mu.RLock()
 		if r, ok := s.resources[uid]; ok {
@@ -210,12 +208,19 @@ func (s *store) onDeleteResource(obj *unstructured.Unstructured) {
 		}
 		s.mu.RUnlock()
 	}
+
 	// Try to determine the application ID by traveling its owners.
 	if appID == "" {
 		s.mu.RLock()
 		appID = s.findAppIDByOwners(owners)
 		s.mu.RUnlock()
 	}
+
+	// This must be done before deleting the resource from the dependedNodes
+	// to ensure that all items in the resources list can be found from one of the app.
+	s.mu.Lock()
+	delete(s.resources, uid)
+	s.mu.Unlock()
 
 	// Delete the resource to the application's dependedNodes.
 	s.mu.RLock()
@@ -226,10 +231,6 @@ func (s *store) onDeleteResource(obj *unstructured.Unstructured) {
 			s.addEvent(event)
 		}
 	}
-
-	s.mu.Lock()
-	delete(s.resources, uid)
-	s.mu.Unlock()
 }
 
 func (s *store) getAppManagingNodes(appID string) map[string]node {

